backend/oqtopus_cloud/user/routers/jobs.py

```diff
@@ -53,27 +57,73 @@ def __init__(self, detail: str):
 
 @router.get(
     "/jobs",
-    response_model=list[JobDef],
+    response_model=list[Union[GetJobsResponse, JobDef]],
     responses={500: {"model": Detail}},
 )
 @tracer.capture_method
 def get_jobs(
     event: Event,
+    fields: Optional[str] = None,
+    startTime: Optional[str] = None,
+    endTime: Optional[str] = None,
+    q: Optional[str] = None,
+    order: Optional[str] = None,
+    size: Optional[str] = None,
+    page: Optional[str] = None,
     db: Session = Depends(get_db),
-) -> list[JobDef] | ErrorResponse:
+) -> list[Union[GetJobsResponse, JobDef]] | ErrorResponse:
     try:
         owner = event.state.owner
         logger.info("invoked!", extra={"owner": owner})
-        stmt = select(Job).filter(Job.owner == owner).order_by(Job.created_at)
-        jobs = db.scalars(stmt).all()
-        results: list[JobDef] = []
-        for job_model, job in [(job, model_to_schema(job)) for job in jobs]:
+
+        # Order Control
+        if order == "ASC" or order is None:
+            arg_order = asc(Job.created_at)
+        elif order == "DESC":
+            arg_order = desc(Job.created_at)
+        else:
+            arg_order = asc(Job.created_at)
+
+        # Fields Control
+        if fields is not None:
+            fields_list = fields.split(",")
+            MAP_SCHEMA_TO_MODEL = {v: k for k, v in MAP_MODEL_TO_SCHEMA.items()}
+            converted_fields_list = [
+                MAP_SCHEMA_TO_MODEL[field] for field in fields_list
+            ]
+            columns = [getattr(Job, field) for field in converted_fields_list]
+            arg_select = columns
+        else:
+            arg_select = [Job]
+
+        stmt = select(*arg_select).filter(Job.owner == owner).order_by(arg_order)
+
+        # Filtering Jobs
+        if startTime is not None:
+            stime = datetime.fromisoformat(startTime).astimezone(jst)
+            stmt = stmt.filter(Job.created_at >= stime)
+        if endTime is not None:
+            etime = datetime.fromisoformat(endTime).astimezone(jst)
+            stmt = stmt.filter(Job.created_at <= etime)
+        if q is not None:
+            stmt = stmt.filter(or_(Job.name.contains(q), Job.description.contains(q)))
+
+        set_params(
+            Params(
+                size=int(size) if size is not None else 100,
+                page=int(page) if page is not None else 1,
+            )
+        )
+        set_page(Page[Job | Row])
+        jobs = paginate(db, stmt)
+
+        results = []
+        for job_model, job in [(job, model_to_schema(job)) for job in jobs.items]:
```

Comment:
You don't have to do this now, but if your code is going to get more complex in the future, you'll want to cut out the query.